pinot-tools/src/main/java/com/linkedin/pinot/tools/segment/converter/ColumnarToStarTreeConverter.java

```diff
@@ -113,7 +115,11 @@ private void convertSegment(File columnarSegment)
     config.setOutDir(_outputDirName);
     config.setStarTreeIndexSpecFile(_starTreeConfigFileName);
     config.setOverwrite(_overwrite);
-    config.setSegmentName(columnarSegment.getName());
+
+    // Read the segment and table name from the segment's metadata.
```

Comment:
Please use segmentNameGenerator instead of setSegmentName. that is the direction in whcih we are evloving